### ISSUE TITLE
Add mutex_m as a gem dependency

### ISSUE THREAD

**Issue by jeremyevans** [minitest/minitest#969]
Starting in Ruby 3.4.0, mutex_m will no longer be in the standard library.  It may be included as a bundled gem, but should be listed as a gem dependency so it can be handled correctly by bundler.  Note that mutex_m is a default gem in Ruby 2.6 (minimum Ruby version for minitest), so this should not cause any backwards compatibility issues.

Here's the warning you get when trying to load minitest in Ruby 3.3.0-preview2:

```
/usr/local/lib/ruby/gems/3.3/gems/minitest-5.20.0/lib/minitest.rb:3: warning: mutex_m which will be not part of the default gems since Ruby 3.4.0
```

**Comment by zenspider** [comment]
this is fixed in 5f5c2111f36658fd2636b108b8327ce4b2f3cf8d

**Comment by fxn** [comment]
The test suite of Zeitwerk [runs against Ruby HEAD every night](https://github.com/fxn/zeitwerk/actions/workflows/keep-an-eye-on-ruby-head.yml?query=branch%3Amain). The last run failed due to this. Would it be possible to cut a release?

**Comment by zenspider** [comment]
@fxn Just released v5.21.0 with this fix.
